proxy/pkg/middleware/public_share_auth.go

```diff
@@ -32,12 +33,23 @@ func PublicShareAuth(opts ...Option) func(next http.Handler) http.Handler {
 				return
 			}
 
-			// We can ignore the username since it is always set to "public" in public shares.
-			_, password, ok := r.BasicAuth()
+			var sharePassword string
+			if signature := r.URL.Query().Get("signature"); signature != "" {
```

Comment:
I think a switch statement would be more readable, especially if more strategies are added ^^ but this is perfectly fine with me and could get merged

Reply:
Yeah, I agree with you. Though I would turn it to a switch statement once we add a new condition.